src/app/components/PasswordField/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { Box, FormField, Button, TextInput, Tip } from 'grommet'
+import { View, Hide } from 'grommet-icons/icons'
+import * as React from 'react'
+
+interface Props {
+  value: string
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
+  error: string | false
+  inputElementId: string
+  placeholder: string
+  showTip: string
+  hideTip: string
+
+  width: string
+}
+
+export function PasswordField(props: Props) {
+  const [passwordIsVisible, setPasswordIsVisible] = React.useState(false)
+
+  return (
+    <FormField
+      htmlFor={props.inputElementId}
+      error={props.error ? props.error : ''}
+      border
+      contentProps={{ border: props.error ? 'bottom' : false }}
+      round="small"
+      width={props.width}
+    >
+      <Box direction="row" align="center">
+        <TextInput
+          id={props.inputElementId}
+          placeholder={props.placeholder}
+          value={props.value}
+          onChange={props.onChange}
+          type={passwordIsVisible ? 'text' : 'password'}
+          plain
+        />
+        <Tip content={passwordIsVisible ? props.hideTip : props.showTip}>
+          <Button
+            onClick={() => setPasswordIsVisible(!passwordIsVisible)}
+            icon={passwordIsVisible ? <View /> : <Hide />}
```

Comment:
I think the icons should be the other way around.
 * When the text is hidden, we want to offer showing in, hence the "view" icon (the eye) should be shown
 * When the text is visible, we want to offer hiding it, hence the "hide" icon should be shown.
 
I tried to look at other implementations, and most are doing it that way.

![image](https://user-images.githubusercontent.com/2093792/191861356-2cfce98e-6e12-448f-a348-cd5ab739b628.png)

Twitter:

![image](https://user-images.githubusercontent.com/2093792/191861719-96ee497f-37c8-42ae-b23a-b659b23e41e9.png)

![image](https://user-images.githubusercontent.com/2093792/191861847-be80b3be-aee3-40f6-ab37-d81000f183e3.png)

BitStamp:

![image](https://user-images.githubusercontent.com/2093792/191862269-d07ad049-b2e8-4d30-befe-95d57f085fea.png)

![image](https://user-images.githubusercontent.com/2093792/191862301-efef9660-3c0f-4c76-85dc-b930ea5c5a70.png)

It's not fully consistent; for example FaceBook has it backwards. 

But I still think that the majority is on this side, and it makes more sense.

Reply:
Moved into https://github.com/oasisprotocol/oasis-wallet-web/issues/1022